### PR TITLE
Fix: do not fail on nested unknown operators

### LINKED ISSUE #10560: Error when nesting an unknown logical operator
**Tell us about your environment**

* **ESLint Version:** 5.0.0
* **Node Version:** 9.11.2

**What parser (default, Babel-ESLint, etc.) are you using?** babel-eslint@9.0.0-beta.1

**Please show your full configuration:**

<details>
<summary>Configuration</summary>

```js
{
  "parser": "babel-eslint",
  "rules": {
    "strict": 0
  }
}
```

</details>

**What did you do? Please include the actual source code causing the issue, as well as the command that you used to run ESLint.**

```js
foo && bar ?? baz
```

```bash
node_modules/.bin/eslint my-file.js
```

**What did you expect to happen?**

No errors

**What actually happened? Please include the actual, raw output from ESLint.**

```
TypeError: Cannot read property 'trueForkContext' of null
    at CodePathState.popChoiceContext (<some-path>/node_modules/eslint/lib/code-path-analysis/code-path-state.js:403:35)
    at processCodePathToExit (<some-path>/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:457:23)
    at CodePathAnalyzer.leaveNode (<some-path>/node_modules/eslint/lib/code-path-analysis/code-path-analyzer.js:653:9)
    at nodeQueue.forEach.traversalInfo (<some-path>/node_modules/eslint/lib/linter.js:863:28)
```

### PR DESCRIPTION
**Pull request by rubennorte**
**What is the purpose of this pull request? (put an "X" next to item)**

[ ] Documentation update
[X] Bug fix ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/bug-report.md))
[ ] New rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-proposal.md))
[ ] Changes an existing rule ([template](https://raw.githubusercontent.com/eslint/eslint/master/templates/rule-change-proposal.md))
[ ] Add autofixing to a rule
[ ] Add a CLI option
[ ] Add something to the core
[ ] Other, please explain:

**What changes did you make? (Give an overview)**

Fixes #10560, after adding support in #10476.

### CODE REVIEW

File: tests/lib/linter.js [hunk 1]
```diff
@@ -4480,6 +4480,16 @@ describe("Linter", () => {
                 assert.strictEqual(messages.length, 0);
             });
 
+            it("should not throw or report errors when the custom parser returns nested unrecognized operators (https://github.com/eslint/eslint/issues/10560)", () => {
+                const code = "null %% 'foo'";
```
Comment (platinumazure) [inline tests/lib/linter.js]: Although the `code` ultimately doesn't matter due to the nature of the parser fixture, could this line be modified to reflect the source code being simulated in the fixture, just for consistency? (`foo && bar ?? baz`)

Comment (rubennorte) [reply tests/lib/linter.js]: I manually modified the fixture so the operator was really `unknown`. I did it that way so when the `??` standardizes we don't have to change this test. Only add it to `isHandledLogicalOperator` and handle it in a couple of places.

Comment (platinumazure) [reply tests/lib/linter.js]: Apologies, I mistyped. I had meant to ask that this source code be used, to match the fixture: `foo && bar %% baz`

My issue is not that you're using an unknown operator, but rather that the `code` variable's value appears to have been copied, pasted, and not modified, compared to the unit test above this one.

Hope this helps and apologies again for the confusion!

Comment (rubennorte) [reply tests/lib/linter.js]: Oh, sorry! My brain parser only saw the change in the operator, not in the variables, haha. I'll change that now. Thanks!

**Comment by rubennorte** [comment]
@platinumazure done. Thanks!

**Pull request merged**